eng/Build.props

```diff
@@ -3,7 +3,7 @@
 <Project>
 
   <PropertyGroup>
-    <RestoreUseStaticGraphEvaluation Condition="'$(DotNetBuildFromSource)' != 'true'">true</RestoreUseStaticGraphEvaluation>
+    <RestoreUseStaticGraphEvaluation>true</RestoreUseStaticGraphEvaluation>
```

Comment:
You’ll need to leave this as-is until your change ships in a build used by Source Build right?

Reply:
If Source Build gets this change in Build.props, wouldn't it also get the fix?

Reply:
Yeah but source build doesn’t use current MSBuild to build (as far as I know). I think it uses our build.cmd which uses a pretty recent version but will lag behind a little until this change is committed and inserted to VS and .NET SDK. Right?

Reply:
I know very little about source build, so I have no idea. Pushed the change 🙂 Thanks!

Reply:
It looks like it runs `build.sh`:
https://github.com/dotnet/msbuild/blob/764b54b73225a14353e2a1a10e72ae8febb779e0/eng/common/templates/steps/source-build.yml#L71

`build.sh` installs the .NET SDK version based on what's in `global.json`:
https://github.com/dotnet/msbuild/blob/764b54b73225a14353e2a1a10e72ae8febb779e0/eng/Versions.props#L34

So until we're using a version of .NET SDK with this change, we'll need to leave Static Graph restore off for Source Build.  Might be worth opening a tracking issue to do that if you haven't already.

Reply:
The version of MSBuild used in source-build isn't directly controlled by our repo at all--it's a property of sourcebuild itself. The job that runs in our PR checks is only a simulation and I don't think it's sufficient on this front to be a good canary for when we're ready to remove this.